src/v/storage/segment.cc

```diff
@@ -530,7 +530,8 @@ ss::future<append_result> segment::do_append(const model::record_batch& b) {
           // inflight index. trimmed on every dma_write in appender
           _inflight.emplace(end_physical_offset, b.last_offset());
           // index the write
-          _idx.maybe_track(b.header(), start_physical_offset);
+          _idx.maybe_track(
+            b.header(), model::timestamp::now(), start_physical_offset);
```

Comment:
Not a fully formed thought yet, but rather than doing this on every append, I wonder if it makes sense to only get the clock timestamp right before the index_state is flushed, provided any user data has been appended since the last flush

Reply:
Interesting idea. i wanted to write a unit test that uses CPU profiling to have a look at this

Reply:
Good flag. We should absolutely ensure Perf impact is studied, understood and if possible, alleviated, here.

Reply:
`seastar::lowres_system_clock` it the way to go. And I think we should also replace `std::chromo::system_clock` with this in `model::timestamp::now`.

Reply:
cc: @StephanDollberg or @ballard26 to advise/approve the performance aspect here.

Reply:
I think it really depends on the usecase. If the lowres is fine then yes go with that but if you need the highres (for example if used for latency measurements) then using that is fine as well.